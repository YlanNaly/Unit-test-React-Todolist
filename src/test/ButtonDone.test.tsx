import React from "react";
import {fireEvent, render} from "@testing-library/react";
import TasksButtonDone from "../todolist/ButtonDone";
import AddTask from "../todolist/Ajout";


it("render input done in snapshot" , ()=>{
    const task = [{ id: 1, text: "Manger!", done: true }]
    const tree = render(<TasksButtonDone deleteTask={jest.fn().mock} task={task} toggleTask={null}/>) ;

    const input = tree.getByTestId("listing");

    expect(input).toMatchSnapshot();
})

it("render onchange method ok" , ()=>{
    const value = {id:1 , text:"mada"};
    const tree = render(<TasksButtonDone task={value}/>);
    const input= tree.getByTestId("listing") as  HTMLInputElement;
    (fireEvent.change(input, {target: {value : value}}));
    expect(input.textContent).toEqual(value.text.concat("Ã—"));
})
//TODO
/*
it("render onchange method ko" , ()=>{
    const value = {id:1 , text:"mada"};
    const tree = render(<TasksButtonDone task={value}/>);
    const input= tree.getByTestId("listing") as  HTMLInputElement;
    const checkbox = tree.getByTestId("checkbox") as HTMLInputElement;
    !checkbox.checked;
    (fireEvent.change(input, {target: {value : value}}));
    expect(input.textContent).toEqual("");
})
*/
//TODO
/*
it("render onClick method " , ()=>{
    const value = {id:1 , text:"mada"};
    const tree = render(<TasksButtonDone task={value} deleteTask={value.id}/>);
    const input= tree.getByTestId("listing") as  HTMLInputElement;
    // @ts-ignore
    fireEvent.change(input.getElementsByClassName("forme").item(0) , "");

    expect(input.textContent).toEqual("");
})
*/
