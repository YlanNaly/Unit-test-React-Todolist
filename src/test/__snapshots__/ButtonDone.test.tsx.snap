// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render input done in snapshot 1`] = `
<li
  class="list-group-item"
  data-testid="listing"
>
  <input
    checked=""
    data-testid="checkbox"
    type="checkbox"
  />
  <span
    class="forme"
    role="button"
    style="padding: 5px; margin-left: 20px; font-size: 20px;"
  >
    Ã—
  </span>
</li>
`;
